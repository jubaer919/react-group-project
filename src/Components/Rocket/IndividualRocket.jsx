import PropType from 'prop-types';
import { useDispatch } from 'react-redux';
import { rocketReserve, cancleReservation } from '../../Redux/Rocket/rocketSlice';
import classes from './IndividualRocket.module.css';

export default function IndividualRocket({
  name, img, des, id, re,
}) {
  const dispatch = useDispatch();
  const reservedHandeler = () => {
    dispatch(rocketReserve(id));
  };

  const cancleHandeler = () => {
    dispatch(cancleReservation(id));
  };

  return (
    <div className={classes.container}>
      <div className={classes['img-container']}>
        <img src={img} alt="rocket" className={classes.img} />
      </div>
      <div className={classes['info-container']}>
        <h1>{ name }</h1>
        <div>
          {re
            && <span className={classes.reserved}>Reserved</span>}
          <span>{des}</span>
        </div>
        <div>
          {!re && <button type="button" className={classes['reserve-btn']} onClick={reservedHandeler}>Reserve Rocket</button>}
          {re && <button type="button" className={classes['cancel-btn']} onClick={cancleHandeler}>Cancell Reservation</button>}
        </div>
      </div>
    </div>
  );
}

IndividualRocket.propTypes = {
  name: PropType.string.isRequired,
  img: PropType.string.isRequired,
  des: PropType.string.isRequired,
  id: PropType.string.isRequired,
  re: PropType.bool.isRequired,
};
