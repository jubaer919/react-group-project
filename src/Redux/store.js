import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './Rocket/rocketSlice';
import missionReducer from './missions/missions';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    missions: missionReducer,
  },
});

export default store;
