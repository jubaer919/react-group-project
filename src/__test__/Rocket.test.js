import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import IndividualRocket from '../Components/Rocket/IndividualRocket';
import store from '../Redux/store';

describe('Test for Rocket', () => {
  test('should render', () => {
    const rocket = {
      id: 1,
      rocketId: 'Rocket',
      image: 'image.png',
      name: 'Rocket',
      description: 'A test rocket',
      reserved: false,
    };
    const tree = renderer.create(
      <Provider store={store}>
        <IndividualRocket
          key={rocket.id}
          name={rocket.name}
          id={rocket.rocketId}
          description={rocket.description}
          image={rocket.image}
          reserved={rocket.reserved}
        />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
