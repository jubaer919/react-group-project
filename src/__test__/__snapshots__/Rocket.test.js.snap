// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Rocket should render 1`] = `
<div
  className="container"
>
  <div
    className="img-container"
  >
    <img
      alt="rocket"
      className="img"
    />
  </div>
  <div
    className="info-container"
  >
    <h1>
      Rocket
    </h1>
    <div>
      <span />
    </div>
    <div>
      <button
        className="reserve-btn"
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
